abstract/group.go

```diff
@@ -51,6 +51,9 @@ type Secret interface {
 
 	// Set to a fresh random or pseudo-random secret
 	Pick(rand cipher.Stream) Secret
+
+	// SetBytes will take bytes and create a secret out of it
+	SetBytes([]byte) Secret
```

Comment:
But big.Int has quite different semantics from nist.Int.  big.Int is supposed to store an *arbitrary* big-integer.  nist.Int is supposed to store a *modular* integer (i.e., an element of a modular integer group or a scalar secret/exponent).  Because of this difference, big.Int by definition never needs to "decide" what to do if an integer supplied to SetBytes is "too big" because no integer is "too big".  nist.Int by contrast does need to decide what to do with a too-big integer, and the most convenient/useful semantics seems to apply the appropriate modulus.

Actually, perhaps some of your discomfort comes from the fact that "nist.Int" is probably misnamed.  For one thing, it should probably not be part of the 'nist' package at all (especially since that package probably should not even be part of the v1.0-stable version of crypto we're preparing), but probably elsewhere - I just haven't quite figured out where yet.  For another thing, perhaps it should not just be named 'Int' but instead perhaps 'ModInt' or something like that, to make it clear that it's not just a "different version of" big.Int but in fact does something else (which is true).